lens/history.go

```diff
@@ -0,0 +1,283 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package lens
+
+import (
+	"context"
+
+	"github.com/ipfs/go-datastore/query"
+	"github.com/sourcenetwork/immutable"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/core"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+// historyItem represents an item in a particular schema's history, it
+// links to the previous and next version items if they exist.
+type historyItem struct {
```

Comment:
suggestion/nitpick: `Link` is a better name imo. Might as well use the full name tho. This is realy a `schemaHistoryLink`. The "schema" prefix is being omitted in this file. Feels important to be specific. If the package was called `schema` then it wouldnt be as important.

Reply:
Will look at changing these (`historyItem`, `targetedHistoryItem`, `getTargetedHistory`, `historyPairing`, `getHistory`).

- [ ] Review history.go names

Reply:
Renamed